modules/core/src/smithy4s/http/UrlForm.scala

```diff
@@ -0,0 +1,146 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+package http
+
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+
+import smithy4s.codecs.PayloadPath
+import smithy4s.codecs.PayloadPath.Segment
+import scala.collection.mutable
+import smithy4s.schema.CachedSchemaCompiler
+import smithy4s.http.internals.UrlFormDataEncoder
+import smithy4s.http.internals.UrlFormDataEncoderSchemaVisitor
+
+import smithy4s.schema.Schema
+
+private[smithy4s] final case class UrlForm(
+    formData: UrlForm.FormData.MultipleValues
+)
+
+private[smithy4s] object UrlForm {
+
+  sealed trait FormData extends Product with Serializable {
+    def prepend(segment: PayloadPath.Segment): FormData
+    def render: String
```

Comment:
I think this should take a `StringBuilder` parameter and imperatively write to it.

Reply:
Nice. Done in https://github.com/disneystreaming/smithy4s/pull/1112/commits/ff5a7c14dd60fc08bf699551c3a8e77e280a8b02.